### ISSUE TITLE
Loss of node attributes

### ISSUE THREAD

**Issue by chads2000** [hbi99/defiant.js#45]
`JSON.toXML(Defiant.node.toJSON(Defiant.xmlFromString('<data><product id="100">MacBook Pro</product><product id="32">Sony Vaio</product></data>')))`

Notice that the id attributes are lost when converting back to XML for String | Number elements

**Comment by hbi99** [comment]
@chads2000 
DefiantJS is targeting JSON as data structure. In order to do that, JSON needs to be transformed...but in a XPath compliant manner. Meaning that, when transformed, the semantics added by the transformation, are added so that they are "invisible". This is accomplished by adding defiant-specific namespace (you can see this if you switch to XML-view in XPath Evaluator). The added namespace "helps" transforming back XML to JSON.

As I explain in the "Caveats" section at [http://defiantjs.com](http://defiantjs.com);
```
Transforming XML to JSON is not lossless since JSON isn't built to match the same 
level of data semantics as XML.
````
This means that XML can not be transformed to JSON lossless...for instance, try transforming XML that contains namespaces - this would result in a incomprehensible soup that will not make any sense. In such scenario, there will be a lot more loss than just attributes.

Since JSON is the targeted data structure, having XML as starting point (as exemplified by you) is the wrong way to go when using DefiantJS.

To summarize:
XML -> JSON == false
JSON -> XML == works
XML (with custom namespace) -> JSON == works
